### DISCUSSION TITLE
How can we verify user and app store event

### DISCUSSION THREAD

**Discussion by vinaye42** [imdhemy/laravel-in-app-purchases#415] [Q&A]
Hi @imdhemy,

Our user has already purchased subscription from app store, but we have not their unique identifier in our back end database,
so how can we connect that users and app store event?

https://developer.apple.com/documentation/appstorereceipts/verifyreceipt
The verifyReceipt endpoint is deprecated. 
Thanks in advanced

**Comment by imdhemy** [comment]
Hi @vinaye42 

You can follow the pinned discussion for ideas on connecting users with IAP in your application. #24 

> The verifyReceipt endpoint is deprecated.

Yes, it is. I'd like to add the new suggested ways from the App Store, but unfortunately, I don't have an Apple Developer Program membership at the moment or at least an application to test with during development. I'd be grateful if you could support me in this regard.

**Reply by vinaye42** [reply]
Hi @imdhemy,
I have a transactionID of user in my users table, but i don't have their transaction detail,
Should we call the below API from laravel code  to  get event from APP STORE?
Or Does App Store send us event or we manually call the below api from our code ?


Production URL: GET https://api.storekit.itunes.apple.com/inApps/v1/transactions/{transactionId}
Sandbox URL: GET https://api.storekit-sandbox.itunes.apple.com/inApps/v1/transactions/{transactionId}

**Reply by vinaye42** [reply]
> Hi @vinaye42
> 
> You can follow the pinned discussion for ideas on connecting users with IAP in your application. #24
> 
> > The verifyReceipt endpoint is deprecated.
> 
> Yes, it is. I'd like to add the new suggested ways from the App Store, but unfortunately, I don't have an Apple Developer Program membership at the moment or at least an application to test with during development. I'd be grateful if you could support me in this regard.


Okay, if we find out the way, i will post here.

**Comment by vinaye42** [comment]
Hello @imdhemy ,
When DID_RENEW event i got different response, $subscription->getProviderRepresentation();

[2025-04-08 04:59:07] production.INFO: renew - object  
[2025-04-08 04:59:07] production.INFO: array (
  'notificationType' => 'DID_RENEW',
  'notificationUUID' => 'cb1fb147-xxxx-xxxx-xxxx-xxxxxxxxx',
  'data' => 
  array (
    'appAppleId' => 1193361491,
    'bundleId' => 'com.fitness',
    'bundleVersion' => '1.0',
    'environment' => 'Production',
    'signedTransactionInfo' => '',
    'signedRenewalInfo' => '',
    'status' => 1,
  ),
  'version' => '2.0',
  'signedDate' => 1744088340641,
)  

[2025-04-08 02:17:33] production.INFO: renew - object  
[2025-04-08 02:17:33] production.INFO: array (
  'quantity' => '1',
  'product_id' => 'monthly',
  'transaction_id' => 'xxxxxxxxx',
  'purchase_date' => '2025-04-08 10:17:28 Etc/GMT',
  'purchase_date_ms' => '1744107448000',
  'purchase_date_pst' => '2025-04-08 03:17:28 America/Los_Angeles',
  'original_purchase_date' => '2025-02-08 11:17:29 Etc/GMT',
  'original_purchase_date_ms' => '1739013449000',
  'original_purchase_date_pst' => '2025-02-08 03:17:29 America/Los_Angeles',
  'expires_date' => '2025-05-08 10:17:28 Etc/GMT',
  'expires_date_ms' => '1746699448000',
  'expires_date_pst' => '2025-05-08 03:17:28 America/Los_Angeles',
  'web_order_line_item_id' => 'xxxxxxxxx',
  'is_trial_period' => 'false',
  'is_in_intro_offer_period' => 'false',
  'original_transaction_id' => 'xxxxxxxx',
  'in_app_ownership_type' => 'PURCHASED',
  'subscription_group_identifier' => 'xxxxxx',
)  

What is this response changes occur, can you help me to fix this issue?

**Comment by bladeodessa** [comment]
Hello @imdhemy !
Is there any news or plans to change the approach of the deprecated verifyReceipt method?
I'm unsure when this API will cease to function, and I have some concerns about deploying this package in production.

**Reply by imdhemy** [reply]
Hello @bladeodessa 
We got [an initiative here](https://github.com/imdhemy/laravel-in-app-purchases/discussions/474) from @TonyHuzz. But the Server server notifications V2 are supported, which Apple lists as[ one of the alternatives](https://developer.apple.com/documentation/appstorereceipts). 

> Receipts are deprecated. To validate In-App Purchases on your server without using receipts, call the [App Store Server API](https://developer.apple.com/documentation/AppStoreServerAPI) to get Apple-signed transaction and subscription information for your customers, or verify the [AppTransaction](https://developer.apple.com/documentation/StoreKit/AppTransaction) and [Transaction](https://developer.apple.com/documentation/StoreKit/Transaction) signed data that your app obtains. You can also get the same signed transaction and subscription information from the [App Store Server Notifications V2](https://developer.apple.com/documentation/AppStoreServerNotifications/App-Store-Server-Notifications-V2) endpoint.
